implementations/rust/ockam/ockam/src/entity/local.rs

```diff
@@ -0,0 +1,219 @@
+use crate::{
+    Contact, ContactsDb, Entity, KeyAttributes, Profile, ProfileAuth, ProfileChangeEvent,
+    ProfileChanges, ProfileContacts, ProfileEventAttributes, ProfileIdentifier, ProfileIdentity,
+    ProfileSecrets, ProfileSync, RemoteEntity, RemoteForwarder, RemoteForwarderInfo, Route,
+    SecureChannelTrait,
+};
+use ockam_core::{Address, AddressSet, Message, Result, Worker};
+use ockam_node::{Cancel, Context};
+use ockam_vault_core::{PublicKey, Secret};
+use ockam_vault_sync_core::Vault;
+
+use ockam_core::hashbrown::HashMap;
+
+pub struct LocalEntity {
+    pub ctx: Context,
+    vault: Address,
+    entity: Entity<ProfileSync>,
```

Comment:
Do we want to rename ProfileSync -> EntitySync?

Reply:
@mrinalwadhwa ?